applications/Melodot.md

```diff
@@ -0,0 +1,238 @@
+2. [Research](https://zerodao.notion.site/Rollup-product-tracking-research-b9448e4464404736858e55972ea12152): We conducted some research on Rollup and data availability layer projects, and you can see the project list here.
+
+## Development Roadmap
+
+### Overview
+
+- **Total Estimated Duration:** 4.5 months
+- **Full-Time Equivalent (FTE):** 1.5
+- **Total Costs:** 28,000 DAI
+
+### Milestone 1 — Erasure coding and KZG
+
+- **Estimated duration:** 1 month
+- **FTE:** 1 
+- **Costs:** 5,000 USD
+
+| Number  | Deliverable               | Specification                                                |
+| ------- | ------------------------- | ------------------------------------------------------------ |
+| **0a.** | License                   | Apache 2.0                                                   |
+| **0b.** | Documentation             | We will provide both inline documentation of the code and a basic tutorial that explains how the new functionality works and how they are used. |
+| **0c.** | Testing and Testing Guide | Unit tests will completely cover the Core functionality to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker                    | We will provide Dockerfiles to demonstrate the full functionality of erasure coding and KZG commitments. |
+| 1.      | melo_erasure_coding       | The core part of the system, including 2D erasure coding and KZG commitment-related primitives and common interfaces. |
+
+### Milestone 2 — Consensus
+
+- **Estimated duration:** 1.5 months
+- **FTE:** 1
+- **Costs:** 7,000 USD
+
+| Number  | Deliverable                    | Specification                                                |
+| ------- | ------------------------------ | ------------------------------------------------------------ |
+| **0a.** | License                        | Apache 2.0                                                   |
+| **0b.** | Documentation                  | We will provide both inline documentation of the code and a basic tutorial that explains how a user can spin up a client, connect to the client management application and data through a browser, and create a local development chain. |
+| **0c.** | Testing and Testing Guide      | Higher-level integration tests and unit tests for all modules. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker                         | We will provide Dockerfiles to start multiple nodes with different roles, create a local test network, and run all integration tests. |
+| 1.      | Substrate pallet: excutive_das | Depends on the `frame-executive` pallet, extending the DAS-related scheduling and data structures. |
+| 2.      | Substrate pallet: system_das   | Depends on the `frame-system` pallet, extending the DAS-related actual operations. |
```

Comment:
The dependents are listed here, but could you expand this to include what functions would be built as part of the pallet?

Reply:
OK, it has been updated.